languages/julia/exercises/concept/boolean-logic/.docs/instructions.md

```diff
@@ -0,0 +1,63 @@
+In this exercise, you'll be implementing the quest logic for a new RPG game a friend is developing. The game's main character is Annalyn, a brave girl with a fierce and loyal pet dog. Unfortunately, disaster strikes, as her best friend was kidnapped while searching for berries in the forest. Annalyn will try to find and free her best friend, optionally taking her dog with her on this quest.
+
+After some time spent following her best friend's trail, she finds the camp in which her best friend is imprisoned. It turns out there are two kidnappers: a mighty knight and a cunning archer.
+
+Having found the kidnappers, Annalyn considers which of the following actions she can engage in:
+
+- _Fast attack_: a fast attack can be made if the knight is sleeping, as it takes time for him to get his armor on, so he will be vulnerable.
+- _Spy_: the group can be spied upon if at least one of them is awake. Otherwise, spying is a waste of time.
+- _Signal prisoner_: the prisoner can be signalled using bird sounds if the prisoner is awake and the archer is sleeping, as archers are trained in bird signaling so they could intercept the message.
+- _Free prisoner_: Annalyn can try sneaking into the camp to free the prisoner, but this tactic will only work if the prisoner is awake and the other two characters are sleeping. If the prisoner is sleeping, they'll be startled by Annalyn's sudden appearance and will awaken the other two characters. The prisoner can also be freed if the archer is sleeping and Annalyn has her pet dog with her, as the knight will be scared by the dog and will withdraw, and the archer can't equip his bow fast enough to prevent the prisoner from being freed.
+
+You have four tasks: to implement the logic for determining if the above actions are available based on the state of the three characters found in the forest and whether Annalyn's pet dog is present or not.
+
+<!-- TODO: I feel like storing the state in a variable is usually not what you'd do in actual Julia code in this situation. I wonder if it would be better to define methods such as isawake(person::T) where {T<:Union{Prisoner, Knight, etc}} instead, and tell the student to not worry about their contents yet. That way, it would also be possible to define the later methods as cansignal(person::T). Boolean logic feels like something that should be covered fairly early on, so we can't expect types to be known at this point.-->
```

Comment:
@cmcaine What are your thoughts on this?

I'm not sure which one I prefer. Maybe this context simply doesn't work for boolean logic in Julia.

Reply:
I don't think we should require custom types.

The exercise is definitely contrived, but that's not necessarily a bad thing. This exercise is just fluff around some simple predicates.

If we found ourselves with a bunch of functions like `can_signal_archer`, `can_signal_prisoner`, etc. then _maybe_ it would make sense to move the type from the name of the func to a parameter, but that's not the case here.

In a real choose-your-own-adventure middleware thing, like `ink`, all of this data would just be in globals or a singleton called something like `prisoner_scene_state`. You probably wouldn't bother fleshing out the enemies with their own types because nothing about them matters beyond whether they are asleep or not.

Reply:
Thanks for the input. I'll leave it as it is in the JS exercise then.